From 581c4f7284efaf8d1831f29c8c07e6f9a5e8d102 Mon Sep 17 00:00:00 2001
From: Joshua Harlow <jxharlow@godaddy.com>
Date: Fri, 16 Sep 2016 17:13:44 -0700
Subject: [PATCH] Disable account rc and ec2 keys/secrets

---
 stack.sh | 30 ------------------------------
 1 file changed, 30 deletions(-)

diff --git a/stack.sh b/stack.sh
index 4e8ab1e..fda76bc 100755
--- a/stack.sh
+++ b/stack.sh
@@ -1175,14 +1175,6 @@ if is_service_enabled g-reg; then
     done
 fi
 
-# Create an access key and secret key for Nova EC2 register image
-if is_service_enabled keystone && is_service_enabled swift3 && is_service_enabled nova; then
-    eval $(openstack ec2 credentials create --user nova --project $SERVICE_TENANT_NAME -f shell -c access -c secret)
-    iniset $NOVA_CONF DEFAULT s3_access_key "$access"
-    iniset $NOVA_CONF DEFAULT s3_secret_key "$secret"
-    iniset $NOVA_CONF DEFAULT s3_affix_tenant "True"
-fi
-
 # Create a randomized default value for the keymgr's fixed_key
 if is_service_enabled nova; then
     iniset $NOVA_CONF keymgr fixed_key $(generate_hex_string 32)
@@ -1247,28 +1239,6 @@ if is_service_enabled heat; then
 fi
 
 
-# Create account rc files
-# =======================
-
-# Creates source able script files for easier user switching.
-# This step also creates certificates for tenants and users,
-# which is helpful in image bundle steps.
-
-if is_service_enabled nova && is_service_enabled keystone; then
-    USERRC_PARAMS="-PA --target-dir $TOP_DIR/accrc"
-
-    if [ -f $SSL_BUNDLE_FILE ]; then
-        USERRC_PARAMS="$USERRC_PARAMS --os-cacert $SSL_BUNDLE_FILE"
-    fi
-
-    if [[ "$HEAT_STANDALONE" = "True" ]]; then
-        USERRC_PARAMS="$USERRC_PARAMS --heat-url http://$HEAT_API_HOST:$HEAT_API_PORT/v1"
-    fi
-
-    $TOP_DIR/tools/create_userrc.sh $USERRC_PARAMS
-fi
-
-
 # Save some values we generated for later use
 save_stackenv
 
-- 
2.7.4 (Apple Git-66)

